packages/accounts-controller/src/utils.ts

```diff
@@ -0,0 +1,54 @@
+import { sha256FromString } from 'ethereumjs-util';
+import { v4 as uuid } from 'uuid';
+
+/**
+ * Returns the name of the keyring type.
+ *
+ * @param keyringType - The type of the keyring.
+ * @returns The name of the keyring type.
+ */
+export function keyringTypeToName(keyringType: string): string {
+  switch (keyringType) {
+    case 'Simple Key Pair': {
+      return 'Account';
+    }
+    case 'HD Key Tree': {
+      return 'Account';
+    }
+    case 'Trezor Hardware': {
+      return 'Trezor';
+    }
+    case 'Ledger Hardware': {
+      return 'Ledger';
+    }
+    case 'Lattice Hardware': {
+      return 'Lattice';
+    }
+    case 'QR Hardware Wallet Device': {
+      return 'QR';
+    }
+    case 'Snap Keyring': {
+      return 'Snap Account';
+    }
+    case 'Custody': {
+      return 'Custody';
+    }
+    default: {
+      throw new Error(`Unknown keyring ${keyringType}`);
+    }
+  }
+}
```

Comment:
Suggestion: Would it make sense to move this logic into `keyring-controller`? A single source of truth for keyring types that could easily be referenced and updated would be much less brittle to keyring changes, and it would allow TypeScript to validate keyring-type inputs.

It does look like that last part (validating against specific keyring types instead of just `string`) will require some changes that might be better implemented in a new PR, including aligning the `type` fields in `InternalAccount` type and `KeyringClass` type.

Here's an outline of what the source of truth in `keyring-controller` would look like. 

```
export enum KeyringType {
  simple = 'Simple Key Pair',
  hd = 'HD Key Tree',
  trezor = 'Trezor Hardware',
  ledger = 'Ledger Hardware',
  lattice = 'Lattice Hardware',
  qr = 'QR Hardware Wallet Device',
  snap = 'Snap Keyring',
  custody = 'Custody',
}

export const keyringTypeToName = {
  'Simple Key Pair': 'Account',
  'HD Key Tree': 'Account',
  'Trezor Hardware': 'Trezor',
  'Ledger Hardware': 'Ledger',
  'Lattice Hardware': 'Lattice',
  'QR Hardware Wallet Device': 'QR',
  'Snap Keyring': 'Snap Account',
  Custody: 'Custody',
  '': 'Unknown Keyring',
} as const;

export type KeyringTypes = keyof typeof keyringTypeToName;

export type KeyringNames =
  (typeof keyringTypeToName)[keyof typeof keyringTypeToName];
```

Reply:
Thanks @MajorLift for the suggestions! I'll make a follow on PR to address the keyring types and update the types in the `InternalAccount`